pages/index.js

```diff
@@ -21,6 +22,9 @@ export default function HomePage({ data, onToggleFavorite, favorites }) {
   const [isSearching, setIsSearching] = useState(false);
   const [isDropdown, setIsDropdown] = useState(false);
 
+  const { mutate } = useSWR("/api/snippets");
+  mutate();
```

Comment:
this is unusual, you should not need to mutate on every re-render, better mutate on an actual event

Reply:
this is because you pass data as a prop, while you should use an useSWR here in this component. This way your whole app on the top level needs to reload and remount in order to update.

just do this:

```suggestion
  const { data } = useSWR("/api/snippets")
```

and remove the data prop you pass.